src/nupic/research/temporal_memory.py

```diff
@@ -123,416 +214,197 @@ def compute(self, activeColumns, learn=True):
       - `activeCells`     (set)
       - `winnerCells`     (set)
       - `activeSegments`  (set)
-      - `predictiveCells` (set)
       - `matchingSegments`(set)
-      - `matchingCells`   (set)
-    """
-    prevPredictiveCells = self.predictiveCells
-    prevActiveSegments = self.activeSegments
-    prevActiveCells = self.activeCells
-    prevWinnerCells = self.winnerCells
-    prevMatchingSegments = self.matchingSegments
-    prevMatchingCells = self.matchingCells
-
-    activeCells = set()
-    winnerCells = set()
-
-    (_activeCells,
-     _winnerCells,
-     predictedActiveColumns,
-     predictedInactiveCells) = self.activateCorrectlyPredictiveCells(
-       prevPredictiveCells,
-       prevMatchingCells,
-       activeColumns)
-
-    activeCells.update(_activeCells)
-    winnerCells.update(_winnerCells)
-
-    (_activeCells,
-     _winnerCells,
-     learningSegments) = self.burstColumns(activeColumns,
-                                           predictedActiveColumns,
-                                           prevActiveCells,
-                                           prevWinnerCells,
-                                           self.connections)
-
-    activeCells.update(_activeCells)
-    winnerCells.update(_winnerCells)
-
-    if learn:
-      self.learnOnSegments(prevActiveSegments,
-                           learningSegments,
-                           prevActiveCells,
-                           winnerCells,
-                           prevWinnerCells,
-                           self.connections,
-                           predictedInactiveCells,
-                           prevMatchingSegments)
+    """
+    prevActiveCells = sorted(self.activeCells)
+    prevWinnerCells = sorted(self.winnerCells)
```

Comment:
Sorting these shouldn't be necessary. The cells / columns are walked in order, so these lists are always sorted.